proposed/2022/Multiple-Equivalent-Framework-Support-TFM-As-Aliases.md

```diff
@@ -0,0 +1,220 @@
+
+NuGet's intra restore caching is based on the existing of a single framework. It is difficult to predict the amount of work that'd be required to implement this correctly, but it'll certainly involve public API changes to NuGet libraries. (NuGet ships libraries, NuGet.exe, VS and dotnet.exe)
+
+### Pack challenges
+
+Multi-targetted pack only really works when each build leg (TargetFramework value) has 1 target location such as `lib/net5.0`.
+Packing projects that do target the same framework multiple may be helpful in some scenarios, so completely blocking it is likely not a great idea, but having a coherent experience would require better understanding of the scenarios at question.
+
+At the minimum, the pack command should be able to detect the scenario at hand and *not succeed queitly* as there's no guarantee what gets packed.
+This scenario would require separate design work.
+
+### dotnet.exe challenges
+
+Today, some values in dotnet.exe work with an aliased `TargetFramework`, such as `build` and `publish`. 
+The expectation is that all the commands support this, but this would require a deeper investigation.
+
+#### dotnet list package
+
+`dotnet list package` has an output that pivots on the effective target framework. This would need to be changed to use the aliased value.
+
+NuGet would need to:
+
+- Display the output with an aliased pivot.
+- Update the machine readable output to handle the new schema, <https://github.com/NuGet/Home/issues/11449>. Given that the machine readable output has not shipped yet, this may be something that can be done ahead of time.
+
+#### Visual Studio challenges
+
+Currently the PM UI does not have any support for multi targeting. This isn't anything new, but it can affect the experience of customers moving to SDK projects for the first time.
+
+Another concern are all the APIs for displaying the dependencies, transitive dependencies and Get Installed Packages APIs in NuGet.VisualStudio.Contracts.
+All of these APIs may require an incremental addition.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+## Rationale and alternatives
+
+- N/A. This is a binary decision to either support or not support this scenario.
+
+## Prior Art
+
+<!-- What prior art, both good and bad are related to this proposal? -->
+<!-- Do other features exist in other ecosystems and what experience have their community had? -->
+<!-- What lessons from other communities can we learn from? -->
+<!-- Are there any resources that are relevant to this proposal? -->
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+## Future Possibilities
+
+<!-- What future possibilities can you think of that this proposal would help with? -->
```

Comment:
Unifying all those json files along with providing a solution for <https://github.com/NuGet/Home/issues/11449> could be side-effect of this refactor.

Reply:
I think in a way, the future possibilities are the same as the motivation part. 

This feature is likely not going to be used in defaults often, but it gives *a lot* of flexibility to power users.

Reply:
I don't think unifying the json files is related to this spec, except that both require changes to the json file schema. Neither change would bring the other one closer to implementation. So, I'm resolving this conversation as out of scope.